### DISCUSSION TITLE
Experience Report

### DISCUSSION THREAD

**Discussion by laheadle** [tamizhvendan/honeyeql#13] [General]
Just want to say I spent the last week using honeyeql and am very grateful for it. Here is an experience report I posted on slack.

Begin quote.

I spent the last week converting my 2000 line biff app from xtdb (xt1) to postgres. Here's a brief experience report . I am a solo entrepreneur with a new business and I aim to launch a scalable web app in the fall. I spent the last 4-5 months building some core app functionality, really enjoying xt1, especially for its pull expressions. I wrote utility functions for doing xt1 transactions. I was happy. But I became concerned with the fact that I would have to migrate my code from xt1, and this eventually led me to move away from it. I didn't want to write and test a bunch of code that I would surely have to throw away. So I had to choose between xt2 and postgres. I made my choice based on risk, with postgres being at the end of the day a safer choice and all in all a fine database. But I knew I would miss those pull expressions and the general ease of using xt1. And I was right. Getting started with postgres was not very fun, and outer joins are no substitute for pull expressions. And then I found honeyeql - https://github.com/tamizhvendan/honeyeql, by @Tamizhvendan S. What a lovely library. It gave me back my pull expressions, and came with a great mutation and filtering library too. It made postgres as fun as xt1, as intelligible, as maintainable, without imposing any weird structure on my data. It just worked. I am a very lucky, and a very happy developer today. I just committed my last change in my transition, my first step into building my business on postgres and honeyeql. I think I made the right choice. Wish me luck.

**Comment by cb-tamizhvendansembiyan** [comment]
I'm glad to know @laheadle, and your message made my day. Feel free to share/discuss if you have any challenges/issues while using it :) 

PS: I am also a solo entrepreneur in the side.

**Comment by laheadle** [comment]
Best of luck with your venture!
